app/src/components/clients.js

```diff
@@ -1,18 +1,95 @@
 const config = require('config');
 
+const acronyms = require('./acronyms');
 const KeyCloakServiceClientManager = require('./keyCloakServiceClientMgr');
 const RealmAdminService = require('./realmAdminSvc');
 
+const { lifecycleService } = require('../services');
+
 const clients = {
+  /**
+   * @function addAppDetails
+   * Add various application details for each service client from db
+   * @param {object} serviceClients An object with all service clients as keys
+   * @returns {object[]} An array of service clients
+   */
+  addAppDetails: serviceClients => {
+    // get a promise for each service client
+    return Promise.all(Object.keys(serviceClients).map(async key => {
+      const sc = serviceClients[key];
+
+      // app details
+      const acronymObj = await acronyms.getAcronym(sc.acronym);
+
+      // if a corresponding acronym in db
+      if (acronymObj) {
+        sc.acronymDetails = acronymObj.dataValues;
+
+        // promotions (from lifecycle table for now)
+        const promotions = await lifecycleService.findLatestPromotions(sc.acronymDetails.acronymId);
```

Comment:
Discussed refactoring this so we're not doing db calls per iteration in the loop. Can live with it for now but let's do a backlog ticket to track a performance enhancement. Might not ever be important but if this ever gets used outside of the admin context, or starts getting slow with rising numbers of clients we'll have to address.

Reply:
thanks for pointing this out. i should have really caught that.